### PR TITLE
Add support for setting ecr permissions

### PR DESCRIPTION
**Pull request by smcavallo**
* **Please check if the PR fulfills these requirements**
- [X ] The commit message describes your change
- [X ] Tests for the changes have been added if possible (for bug fixes / features)
- [X ] Docs have been added / updated (for bug fixes / features)
- [X ] Changes are mentioned in the changelog (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature - adds support for setting ecr permissions via ecrSetRepositoryPolicy

* **What is the current behavior?** (You can also link to an open issue here)
feature does not exist

* **What is the new behavior (if this is a feature change)?**
allow managing ecr permissions/policy

* **Does this PR introduce a breaking change?** (What changes might users need to make in their setup due to this PR?)
no

* **Other information**:

### CODE REVIEW

**Comment by hoegertn** [comment]
Thanks for the PR. Can you elaborate on why you would want to set this using the plugin and not using Infra as Code?

**Comment by smcavallo** [comment]
hi @hoegertn 
this was modelled after [ec2ShareAmi](https://github.com/jenkinsci/pipeline-aws-plugin#ec2ShareAmi).

* AMI -> ci creates a new ami, publishes it, and then ensures consumers have access. 
* ECR -> ci builds a new container, publishes it to the registry, ensure consumers have access.

I know AMIs permissions are at the artifact level and ecr permissions are at the repo level.  However, this seemed like a close enough use case.
I agree, using terraform or cloudformation would be a better place to define this, especially since the permissions are "owned" by the ec2 repository, and technically jenkins should be the CI piece and tf/cf would better for CD.

In effect though it, it does provide an extremely simple way to define ecr repo json policy as infrasturcture as code, which is easy to define/implement/maintain:
```
def policyFile ="${env.WORKSPACE}/policyText_with_all_my_aws_accounts.json"
def json = new File(policyFile).text
def id = 'aws-accountId'
ecrSetRepositoryPolicy(registryId: id, repositoryName: 'microservice_1', policyText: json)
ecrSetRepositoryPolicy(registryId: id, repositoryName: 'microservice_2', policyText: json)
ecrSetRepositoryPolicy(registryId: id, repositoryName: 'microservice_3', policyText: json)
ecrSetRepositoryPolicy(registryId: id, repositoryName: 'microservice_4', policyText: json)
ecrSetRepositoryPolicy(registryId: id, repositoryName: 'microservice_5', policyText: json)
```

File: pom.xml [hunk 1]
```diff
@@ -102,7 +102,6 @@
               <encoding>UTF-8</encoding>
               <consoleOutput>true</consoleOutput>
               <failsOnError>true</failsOnError>
-              <linkXRef>false</linkXRef>
```
Comment (hoegertn) [inline pom.xml]: Why?

Comment (smcavallo) [reply pom.xml]: @hoegertn - I have reverted this change.  (It was at the suggestion of my IDE.)

**Comment by smcavallo** [comment]
> 
> 
> I am willing to merge this. Can you please rebase and revert the pom.xml change?

@hoegertn - thank you so much!  I have merged the upstream master and updated the pom.

**Pull request merged**